src/modules/flow/console/console.c

```diff
@@ -45,70 +45,73 @@ struct console_data {
     bool flush;
 };
 
-SOL_ATTR_PRINTF(2, 3) static void
-console_output(struct console_data *mdata, const char *fmt, ...)
+SOL_ATTR_PRINTF(3, 4) static void
+console_output(struct console_data *mdata, bool add_affix, const char *fmt, ...)
 {
     va_list ap;
 
-    fputs(mdata->prefix, mdata->fp);
+    if (add_affix)
+        fputs(mdata->prefix, mdata->fp);
```

Comment:
it's better to give `const char *prefix` and `const char *suffix` as parameters to `console_output()`, that way you can give no prefix ("") or comma/newline for suffix when doing composed.

I also think that `", "` is better than `"\n"` for composed.